plugin/db/util/driverutil.go

```diff
@@ -506,3 +506,28 @@ func fromStoredVersion(storedVersion string) (bool, string, string, error) {
 	}
 	return true, fmt.Sprintf("%d.%d.%d", major, minor, patch), suffix, nil
 }
+
+// GetPITRDatabaseName composes a pitr database name
+func GetPITRDatabaseName(database string) string {
+	now := time.Now().Unix()
+	postfix := fmt.Sprintf("_pitr_%d", now)
+	ret := database + postfix
+	// MySQL max database length is 64
+	if len(ret) > 64 {
```

Comment:
It's ok for a user to give a database name of 64, then we should only truncate it to generate the pitr database name. It's no big deal because we will finally swap it back.

However, if we want to ensure that the generated pitr database name <= 64, the original must be <= 48, which seems unreasonable, especially for those who don't want to use the PITR feature, or import databases before using Bytebase and exceeds 48 characters.

Reply:
Yeah, in this case, I might prefer following gh-ost.

https://sourcegraph.com/github.com/github/gh-ost/-/blob/go/base/context.go?L320&subtree=true